services/web/server/src/simcore_service_webserver/projects/projects_handlers_crud.py

```diff
@@ -212,19 +210,25 @@ async def _copy_files_from_source_project(
             await stack.enter_async_context(
                 projects_api.lock_with_notification(
                     app,
-                    query_params.from_study,
+                    source_project["uuid"],
                     ProjectStatus.CLONING,
                     user_id,
                     await get_user_name(app, user_id),
                 )
             )
-        await clone_data_task
-
-    # unhide the project if needed since it is now complete
-    if not new_project_was_hidden_before_data_was_copied:
-        await db.update_project_without_checking_permissions(
-            new_project, new_project["uuid"], hidden=False
-        )
+        starting_value = task_progress.percent
+        async for long_running_task in copy_data_folders_from_project(
```

Comment:
Just to see if i understood: here you are connecting two long-running tasks

So this mean that the webserver waits (i.e. polls storage until tasks is completed) until the all the data is fully copied. since this coroutine is awaited within ``create_project``. Then the webserver handler for CREATE project is built as a long running task as well.

Reply:
yes 2 long running tasks are connected here.

technically the create_project was **already** waiting for storage to complete. The long running task allows:
- no super large request anymore (the request can spend several minutes)
- it is potentially possible to retry in case of network issue (retrials, this is not yet the case in the frontend, but the client does retry on connection lost (60 seconds)
- progress reporting is now possible (and is implemented as such)
- potentially could also return faster if the request completed fast (not implemented)